--- a/drivers/tty/serial/Kconfig
+++ b/drivers/tty/serial/Kconfig
@@ -1610,5 +1610,5 @@ config SERIAL_ESP32_ACM
 	tristate "Espressif ESP32 USB ACM gadget support"
-	depends on XTENSA_PLATFORM_ESP32 || (COMPILE_TEST && OF)
+	depends on ARCH_ESPRESSIF || XTENSA_PLATFORM_ESP32 || (COMPILE_TEST && OF)
 	select SERIAL_CORE
 	select SERIAL_CORE_CONSOLE
 	select SERIAL_EARLYCON
@@ -1616,6 +1616,6 @@ config SERIAL_ESP32_ACM
 	help
-	  Driver for the CDC ACM gadget controller of the Espressif ESP32S3
-	  SoCs that share separate USB controller with the JTAG adapter.
+	  Driver for the CDC ACM gadget controller of Espressif ESP32 SoCs
+	  that share a separate USB controller with the JTAG adapter.
 	  When earlycon option is enabled the following kernel command line
 	  snippet may be used:
 	    earlycon=esp32s3acm,mmio32,0x60038000
--- a/drivers/tty/serial/esp32_acm.c
+++ b/drivers/tty/serial/esp32_acm.c
@@ -42,1 +42,3 @@
-	}, { /* sentinel */ }
+	}, {
+		.compatible = "espressif,esp32p4-usb-serial-jtag",
+	}, { /* sentinel */ }
